### PR TITLE
Newmodule ultraplex

### LINKED ISSUE #3318: new module: ULTRAPLEX
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by CharlotteAnne**
## PR checklist
Adding new module Ultraplex
Closes #3318 (https://github.com/nf-core/modules/issues/3318)

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/config/pytest_modules.yml [hunk 1]
```diff
@@ -82,14 +82,14 @@ angsd/docounts:
   - modules/nf-core/angsd/docounts/**
   - tests/modules/nf-core/angsd/docounts/**
 
-annotsv/installannotations:
```
Comment (charles-plessy) [inline tests/config/pytest_modules.yml:85]: You can avoid committing these chunks by using the `-p` option of `git add`.

**Review by charles-plessy — APPROVED** [review]
The module looks good to me.  The commit on `pytest_modules.yml` is noisy, but I see that such noise has been tolerated at least once in the past (ba1ad555be19cb6cedb52c49b21e267b0b7110b7).  This said, it would be neat if it would be fixed before the merge.

File: modules/nf-core/ultraplex/meta.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: "ultraplex"
+
+description: Ultraplex is an all-in-one software package for processing and demultiplexing fastq files.
+keywords:
+  - demultiplexer
```
[the remark this answers is not in the document]
Comment (CharlotteAnne) [reply modules/nf-core/ultraplex/meta.yml]: done!

File: tests/config/pytest_modules.yml (hunk 1 above)
Comment (CharlotteAnne) [reply tests/config/pytest_modules.yml]: done!

**Pull request merged**